croptalk/tools.py

```diff
@@ -125,6 +150,94 @@ def get_sp_document(state: Optional[str] = None,
         return missing_var_msg
 
 
+class SQLHandler(BaseCallbackHandler):
+    def __init__(self):
+        self.sql_result = []
+
+    def on_agent_action(self, action, **kwargs):
+        """Run on agent action. if the tool being used is sql_db_query,
+         it means we're submitting the sql and we can
+         record it as the final sql"""
+
+        if action.tool in ["sql_db_query_checker", "sql_db_query"]:
```

Comment:
out of curiosity, what are these 2 names matching exactly? I was expecting them to match tool names we created but it doesn't appear to be the case...

Reply:
this is used to match internal tools used by the SQLAgentExecutor. It's only meant to return the exact query used by the agent and not any tools we developped